examples/LPBatching/lpBatching.go

```diff
@@ -0,0 +1,159 @@
+		Host:     url,
+		Token:    token,
+		Database: database,
+	})
+
+	if cErr != nil {
+		panic(cErr)
+	}
+
+	defer func(client *influxdb3.Client) {
+		err := client.Close()
+		if err != nil {
+			slog.Error("Error closing client", err)
+		}
+	}(client)
+
+	// SYNC WRITE BATCHES
+
+	// create a new Line Protocol Batcher
+	syncLpb := batching.NewLPBatcher(batching.WithBufferSize(4096)) // Set buffer size
+	t := time.Now().Add(-LineCount * time.Second)
+	var wErr error
+
+	// add new data into the batcher
+	for range LineCount {
+		syncLpb.Add(fmt.Sprintf(dataTemplate,
+			ids[rnd.Intn(len(ids))],
+			syncLocations[rnd.Intn(len(syncLocations))],
+			rnd.Float64()*100,
+			rnd.Float64()*360,
+			rnd.Intn(100),
+			t.UnixNano(),
+		))
+		t = t.Add(time.Second)
+
+		// if ready state reached, emit a batch
+		if syncLpb.Ready() {
+			wErr = client.Write(context.Background(), syncLpb.Emit())
+			if wErr != nil {
+				slog.Error(wErr.Error())
+			}
+		}
+	}
+
+	// Write final batch to client
+	wErr = client.Write(context.Background(), syncLpb.Emit())
+	if wErr != nil {
+		slog.Error(wErr.Error())
+	}
+	fmt.Printf("Sync Writes Done.  %d Bytes remaining in batcher buffer\n",
+		syncLpb.CurrentLoadSize())
+
+	//prepare SQL query
+	queryTemplate := "SELECT * FROM \"uasv\"" +
+		"\nWHERE time >= now() - interval '2 minutes'\n" +
+		"AND location IN ('%s','%s')\norder by time"
+
+	fmt.Println("Show selected values written by sync")
+	DumpQueryResults(client.Query(context.Background(),
+		fmt.Sprintf(queryTemplate, syncLocations[2], syncLocations[3])))
+
+	fmt.Println()
+
+	// ASYNC WRITE BATCHES
+	asyncLpb := batching.NewLPBatcher(batching.WithBufferSize(4096), // Set buffer size
+		batching.WithByteEmitReadyCallback(func() { fmt.Println("|-- ready to emit -->") }), // Set ready callback
+		batching.WithEmitBytesCallback(func(bytes []byte) { // Set Callback to handle emitted bytes
+			wErr = client.Write(context.Background(), bytes)
+			if wErr != nil {
+				slog.Error(wErr.Error())
+			}
+		}))
+
+	asyncLocations := []string{"ibiza", "dubai", "phuket", "maui"}
+	t = time.Now().Add(-LineCount * time.Second)
+
+	// Add new data to Batcher
+	for range LineCount {
+		asyncLpb.Add(fmt.Sprintf(dataTemplate, ids[rnd.Intn(len(ids))],
+			asyncLocations[rnd.Intn(len(asyncLocations))],
+			rnd.Float64()*100,
+			rnd.Float64()*360,
+			rnd.Intn(100),
+			t.UnixNano()))
+		t = t.Add(time.Second)
+	}
+
+	// Write the remaining batch records to the client
+	wErr = client.Write(context.Background(), asyncLpb.Emit())
+	if wErr != nil {
+		slog.Error(wErr.Error())
+	}
+	fmt.Printf("Async Writes Done.  %d Bytes remaining in batcher buffer\n",
+		asyncLpb.CurrentLoadSize())
+
+	fmt.Println("Show selected values written by async")
+	DumpQueryResults(client.Query(context.Background(),
+		fmt.Sprintf(queryTemplate, asyncLocations[0], asyncLocations[1])))
+
+}
+
+func DumpQueryResults(qiter *influxdb3.QueryIterator, err error) {
```

Comment:
Why is this extracted to a public function?

Reply:
The example dumps  the data after and sync call and then after the async call, so it gets called twice.  Also I wanted to introduce a method that handled both return types of query, so that the query could be made at the top of the call stack.